src/platform_linux/platform.h

```diff
@@ -368,13 +371,13 @@ extern platform_heap_id     Heap_id;
    ({                                                                          \
       debug_assert((n) >= sizeof(*(v)));                                       \
       (typeof(v))platform_aligned_malloc(                                      \
-         hid, (a), (n), STRINGIFY(v), __func__, __FILE__, __LINE__);           \
+         hid, (a), (n), NULL, STRINGIFY(v), __func__, __FILE__, __LINE__);     \
```

Comment:
For most of the consumers, this is good-enough. I could have changed this to require all callers to also declare an on-stack `platform_memfrag{}`, but that would be more code changes.

The one 'minor' issue with this is that we might incorrectly free a smaller-sized fragment . But that's not a huge loss, so I went with current solution.